src/lib/services.ts

```diff
@@ -0,0 +1,26 @@
+import { eventBus } from "@/utils/eventBus";
+import { monthService } from "@/lib/monthService";
+import { BUDGET_EVENTS } from "../utils/eventTypes";
+
+export function initializeEventListeners() {
```

Comment:
I don't see where do you call this function - initializeEventListeners- probably you should create some hook and call it in some component.

Also you faced here with cascades - some events could trigger some other. If it is not handled properly - it become a nightmare! Lets develop some usage patterns and describe it somewhere (at least in comments if it is not possible to do with the code).